deepchecks/vision/dataset.py

```diff
@@ -81,31 +80,33 @@ class VisionDataset:
     """
 
     _data: DataLoader = None
+    label_transformer: BaseLabelEncoder = None
 
     def __init__(self,
                  data_loader: DataLoader,
-                 label_type: str,
                  num_classes: Optional[int] = None,
-                 label_transformer: Optional[Callable] = None,
+                 label_transformer: Union[ClassificationLabelEncoder, DetectionLabelEncoder] = None,
                  sample_size: int = 1000,
                  sample_iteration_limit: int = 1_000_000,
                  random_seed: int = 0):
         self._data = data_loader
 
         if label_transformer is None:
-            self.label_transformer = lambda x: x
+            self.label_transformer = ClassificationLabelEncoder(lambda x: x)
```

Comment:
Think it's clearer if label_transformer =None will mean no task and only integrity / data checks can run